session/nontransactional.go

```diff
@@ -76,6 +76,13 @@ func (j job) String(redacted bool) string {
 
 // HandleNonTransactionalDML is the entry point for a non-transactional DML statement
 func HandleNonTransactionalDML(ctx context.Context, stmt *ast.NonTransactionalDMLStmt, se Session) (sqlexec.RecordSet, error) {
+	// disable staleread for NT transactions because they always write.
+	sessVars := se.GetSessionVars()
+	stale := sessVars.ReadStaleness
+	sessVars.ReadStaleness = 0
+	defer func() {
+		sessVars.ReadStaleness = stale
```

Comment:
Staleness(and select limit) was handled around line 410. Shall we move them here?

Reply:
I think so, before disabling stale read, it went wrong when preprocessing in L86.